### PR TITLE
Improve sharing buttons events performance

### LINKED ISSUE #34581: Sharing Block: Improve buttons event handling performance
Currently, we are adding event listener to every Sharing Button on the page. In order to have a better performance we want to revert logic to being subscribed to `document` and check the event target.

----

Epic: #19694

### PR DESCRIPTION
**Pull request by andrii-lysenko**
Fixes https://github.com/Automattic/jetpack/issues/34581

## Proposed changes:

Instead of having listener for each button, we are now have only one on parent element. Additionally:
- Adding `id` to parent element
- Adding `service` dataset link item for easier retrieval

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

- Spin up your site with local changes
- Make sure you have beta block enabled. In your environment (example Docker) wp-config.php you should have define( 'JETPACK_BLOCKS_VARIATION', 'beta' );
- Add Sharing Buttons block to a post and check it's behaviour.
- View the post and check Sharing Buttons functionality.

### CODE REVIEW

File: projects/plugins/jetpack/extensions/blocks/sharing-button/sharing-button.php [hunk 1]
```diff
@@ -65,7 +65,8 @@ function render_block( $attr, $content, $block ) {
 	Jetpack_Gutenberg::load_assets_as_required( __DIR__ );
 
 	$component  = '<li class="jetpack-sharing-button__list-item">';
-	$component .= '<a rel="nofollow noopener noreferrer" class="' . $sharing_link_class . '" href="' . $link_url . '" target="_blank" data-shared="' . $data_shared . '" aria-label="' . $link_aria_label . '" primary>';
+	$component .= '<a rel="nofollow noopener noreferrer" class="' . $sharing_link_class . '" href="' . $link_url . '" target="_blank" ';
```
Comment (monsieur-z) [inline projects/plugins/jetpack/extensions/blocks/sharing-button/sharing-button.php]: We could escape attributes here to be on the safer side, but maybe that's me being over-cautious and now fully aware of the data flow here.

**Review by monsieur-z — DISMISSED** [review]
Looks good!

I left a comment you may address or discard.

File: projects/plugins/jetpack/extensions/blocks/sharing-button/sharing-button.php (hunk 1 above)
Comment (andrii-lysenko) [reply projects/plugins/jetpack/extensions/blocks/sharing-button/sharing-button.php]: Thanks! Thats a good point. Do you suggest using `esc_html ` here for params?

**Comment by andrii-lysenko** [comment]
@monsieur-z Thanks for suggestion. Updated.

File: projects/plugins/jetpack/extensions/blocks/sharing-button/sharing-button.php (hunk 1 above)
Comment (monsieur-z) [reply projects/plugins/jetpack/extensions/blocks/sharing-button/sharing-button.php]: `esc_attr` for attributes, including `aria-label`.

**Review by monsieur-z — APPROVED** [review]

**Pull request merged**